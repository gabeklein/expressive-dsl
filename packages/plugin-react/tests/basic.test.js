const test = require("./_adapter");

describe("basics", () => {
  test("component", `
    const Component = () => do {
      <div>Hello World</div>
    }
  `);

  test("multiple", `
    const Component = ({ hello }) => do {
      <this>
        <div>Foo</div>
        <div>Bar</div>
      </this>
    }
  `);

  test("switch", `
    const Component = ({ hello }) => do {
      if(hello)
        <div>Hello World</div>
      else
        <div>Goodbye World</div>
    }
  `);
})