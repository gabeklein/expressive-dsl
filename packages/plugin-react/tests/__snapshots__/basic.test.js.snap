// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic component 1`] = `
"const {
  createElement: create
} = require(\\"react\\");

const Component = () => create(\\"div\\", {}, \\"Hello World\\");"
`;

exports[`basic component 2`] = `
"import React from \\"react\\";

const Component = () => <div>Hello World</div>;"
`;
